src/query/ast/src/parser/expr.rs

```diff
@@ -1110,11 +1110,17 @@ pub fn type_name(i: Input) -> IResult<TypeName> {
     );
     let ty_float32 = value(TypeName::Float32, rule! { FLOAT32 | FLOAT });
     let ty_float64 = value(TypeName::Float64, rule! { FLOAT64 | DOUBLE });
-    let ty_decimal = map(
+    let ty_decimal = map_res(
         rule! { DECIMAL ~ "(" ~ #literal_u64 ~ "," ~ #literal_u64 ~ ")" },
-        |(_, _, precision, _, scale, _)| TypeName::Decimal {
-            precision: precision as u8,
-            scale: scale as u8,
+        |(_, _, precision, _, scale, _)| {
+            Ok(TypeName::Decimal {
```

Comment:
No validation here ？

Reply:
the precise validation should be in DecimalDataType::from_size()